### ISSUE TITLE
Introduce GigE Vision Control Protocol (GVCP) protocol

### ISSUE THREAD

**Issue by tigercosmos** [seladb/PcapPlusPlus#1321]
GigE Vision[[1]](https://en.wikipedia.org/wiki/GigE_Vision#cite_note-1) is an [interface standard](https://en.wikipedia.org/wiki/Interface_standard) introduced in 2006 for high-performance industrial cameras.

To support it, we will add a new layer beyond the UDP layer.

There are plenty of commands in GVCP, the first step would be:
- recognized UDP port 3956 as GVCP
- parse GVCP command header
- parse GVCP acknowledge header

Fully support of all commands and their response could be the next step.

see more references:
- Wikipedia: https://en.wikipedia.org/wiki/GigE_Vision
- Wireshark: https://www.wireshark.org/docs/dfref/g/gvcp.html
- Related repos: 
  - https://github.com/Touseefelahi/GigeVision
  - https://github.com/AravisProject/aravis
- Spec file: https://www.automate.org/vision/vision-standards/download-the-gige-vision-standard

**Comment by seladb** [comment]
@tigercosmos are you planning to work on it?

**Comment by tigercosmos** [comment]
> @tigercosmos are you planning to work on it?

Yes, but I can expect this to be a big PR. I will complete it little by little.

**Comment by tigercosmos** [comment]
@seladb according to my experience, full support may take about a few thousand lines of code. Do you agree that we separate it into several PRs? maybe 2 or 3.

**Comment by seladb** [comment]
@tigercosmos sure, we can start by basic support and add more later

**Comment by seladb** [comment]
@tigercosmos should we keep this issue or close it until we see more interest?

**Comment by tigercosmos** [comment]
I guess we can open a new issue for tracking all protocols to be implemented.

**Comment by seladb** [comment]
GitHub allows opening sub-issues now. Maybe we can move all of these to a parent issue called "Protocol requests."

https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/adding-sub-issues
